src/components/Dashboard/MainInput/MainInput.jsx

```diff
@@ -46,7 +46,7 @@ export default class MainInput extends React.Component {
               value={newTaskVal}
             />
             <InputGroup.Append>
-              <SpeechRecognition resultCallback={this.setInputValue}/>
+              <SpeechRecognition onResultReady={this.setInputValue}/>
```

Comment:
Still not clear what this function is intended to do. Please choose names unambiguously.